lib/relooper/src/graph/cfg/mod.rs

```diff
@@ -35,6 +44,50 @@ impl<TLabel> CfgEdge<TLabel> {
     }
 }
 
+/// A struct which enables iterating over the nodes that make up a `CfgEdge`.
+/// Internally it stores the data as a 2-array as opposed to a `Vec` to avoid heap allocation.
```

Comment:
Out of curiosity, what about small-vector-optimization in rust? Afaik in C++ if size of vector is less than 20 or smth like that it stored on stack.